source/elements/oneMKL/source/domains/spblas/matrix_view.rst

```diff
@@ -0,0 +1,134 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_matrix_view_and_descr:
+
+Matrix view
+===========
+
+.. _onemkl_sparse_matrix_descriptor:
+
+matrix_descr
+------------
+
+.. rubric:: Definition
+
+.. code:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       enum class matrix_descr {
+           general,
+           symmetric,
+           hermitian,
+           triangular,
+           diagonal,
+       };
+
+   }
+
+.. container:: section
+
+   .. rubric:: Description
+
+   The matrix descriptor is an optional property that can optimize some
```

Comment:
It's not a property, it's a "view" in which the APIs will only focus on parts of the user data, and ignore parts that are not part of the view. E.g., the matrix may be non-symmetric, but the "triangular" view using the "lower" triangular region means the API being called (e.g., `spsv`) is to perform operations using only the lower triangular region in the matrix data and to ignore the upper triangular region of the data,

It's actually "properties" that are optional, and "views" (`matrix_descr`) that are required in API arguments.

If users don't specify the "sorted"ness property through the `set_matrix_property()` API, for instance, then the implementation must assume that the data is not sorted, in which case a worse algorithm may need to be used to perform operations.

Reply:
Yes, I understand. I find this type difficult to describe. At the time of writing I thought of "optional" because in some cases this field can only be thought as an optimization i.e. if the full matrix is already triangular using `general` or `triangular` does not affect the output.
I've rephrased this in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/90c4cec984f71f62b1f60d19080cabd3ae643408

Reply:
@gajanan-choudhary can this one be resolved?